extension/api.js

```diff
@@ -78,14 +105,174 @@ const lostEngines = new Map ([
   ["{39790485-930b-40a5-8268-69222363ff80}", Oct2020],
   ["{446c7519-9e32-4f9a-b562-447f4421ec9a}", Oct2020],
   ["{b30e775a-7a10-480d-ace4-761b9ca07aee}", Dec2020],  // Test add-on
+  // ["{fd299ce1-1602-4490-b659-f45504f9324c}", Date.now()], // Manual test addon for dev.
 ]);
 
+// If the extension is not running or enabled in the search service
+// we will try again later.
+let tryagain = false;
+
+function finish(result = false) {
+  Services.prefs.setBoolPref(RUN_ONCE_PREF, result);
+}
+
+async function tryDefaultReset() {
+  let defaultEngine = await Services.search.getDefault();
+  if (lostEngines.has(defaultEngine?._extensionID)) {
+    console.log(`reset-default-search: The default engine was already selected by the user. ${defaultEngine._extensionID}`);
+    finish(true);
+    return;
+  }
+
+  // Make sure the setting is not user controlled.
+  let control = await ExtensionSettingsStore.getLevelOfControl(
```

Comment:
Move this logic after the part that looks up installed add-ons, to avoid the need to access `ExtensionSettingsStore` if (in the common case) the user hasn't got any recognized lost engine.